src/main/java/io/lettuce/core/protocol/DefaultEndpoint.java

```diff
@@ -452,6 +450,7 @@ public void notifyChannelActive(Channel channel) {
         }
 
         sharedLock.doExclusive(() -> {
+            this.channel = channel;
```

Comment:
If you roll-back all the other changes and leave out only this one then your test case still succeeds.

Can you elaborate on why you need them too?

Reply:
Just tested again and I am not sure this is correct.
I will do some additional polishing and accept this change